### PR TITLE
test(AHWR-1756): add unit and integration tests for poultry MI events and bug fix

### PR DESCRIPTION
**Pull request by shahnaazr**
## Summary
- Fix feature-toggle config returning raw config instead of Joi validated result (Bug 1)
- Fix undefined being written to CSV when event payload is invalid (Bug 2)
- Add unit tests for poultry field column mapping and date toggle boundary
- Add integration tests for full pipeline end-to-end against Azurite

## Test plan
- [x] npm test passes (131 unit tests)
- [x] npm run test:integration passes (10 integration tests) — requires Docker and .env setup
- [x] No secrets committed — Azurite key read from .env (gitignored)

Jira: AHWR-1756

### CODE REVIEW

**Review by MiyamotoAkira — APPROVED** [review]

File: test/integration/azurite-config.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const host = process.env.AZURITE_HOST || '127.0.0.1'
+const blobPort = process.env.AZURITE_BLOB_PORT || '10005'
+const tablePort = process.env.AZURITE_TABLE_PORT || '10007'
+const account = process.env.AZURITE_ACCOUNT || 'devstoreaccount1'
```
Comment (codeinabox) [inline test/integration/azurite-config.js]:
```suggestion
const host = process.env.AZURITE_HOST ?? '127.0.0.1'
const blobPort = process.env.AZURITE_BLOB_PORT ?? '10005'
const tablePort = process.env.AZURITE_TABLE_PORT ?? '10007'
const account = process.env.AZURITE_ACCOUNT ?? 'devstoreaccount1'
```

File: ffc-ahwr-mi-reporting/storage/storage.js [hunk 2]
```diff
@@ -76,6 +76,9 @@ const processEntitiesByTimestampPaged = async (fileName, context) => {
 
       try {
         const csvRow = transformEventToCsvV3(event, context)
+        if (csvRow === undefined) {
+          continue
+        }
```
Comment (codeinabox) [inline ffc-ahwr-mi-reporting/storage/storage.js]: Do we have a scenario where this would happen?
